.github/workflows/ci.yml

```
@@ -2,66 +2,53 @@ name: CI
 
 on:
   push:
-    branches: [ master ]
+    branches: [master]
   pull_request:
-    branches: [ master ]
+    branches: [master]
+
+permissions:
+  contents: read
+  actions: read
+  checks: write
 
 jobs:
   test:
```

I think the job name is missleading
```suggestion
  setup:
```